### PR TITLE
TD-912-Modify SelfAssessmentResults to link to DelegateUser instead of CandidateID

### PR DESCRIPTION
**Pull request by Auldrin-Possa**
JIRA link
https://hee-tis.atlassian.net/browse/TD-912

### Description
DelegateUserId column added to SelfAssessmentResults populated with userId based on CandidateID
Code modified to replace CandidateID

**Screenshots**
Updated in Jira ticket.

-----
**Developer checks**

I have:
- [ ] Run the formatter and made sure there are no IDE errors
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin]
- [ ] Updated/added documentation in [Confluence]
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Migrations/202301101342_AddSelfAssessmentResultDelegateUserID.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        }
+
+        public override void Down()
+        {
+            Delete.ForeignKey("FK_SelfAssessmentResults_DelegateUserID_Users_ID").OnTable("SelfAssessmentResults");
+            Delete.Column("DelegateUserID").FromTable("SelfAssessmentResults");
+
+            Rename.Column("CandidateID_deprecated").OnTable("SelfAssessmentResults").To("CandidateID");
+            Create.ForeignKey("FK_SelfAssessmentResults_CandidateID_Candidates_CandidateID")
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data.Migrations/202301101342_AddSelfAssessmentResultDelegateUserID.cs:31]: This down migration is likely to fail once we have records with null CandidateID_deprecated. To address this we would need to either populate the records with NULLs or delete them.

**Review by kevwhitt-hee — COMMENTED** [review]
Please check the comment against the down migration. Can we introduce a step to ensure it runs when there are records that have been inserted after the up migration? If not, happy to merge.

**Comment by kevwhitt-hee** [comment]
Please also take a look at the merge conflict (probably changes from TD-262 which has now been merged that are conflicting)

File: DigitalLearningSolutions.Data.Migrations/202301101342_AddSelfAssessmentResultDelegateUserID.cs (hunk 1 above)
Comment (Auldrin-Possa) [reply DigitalLearningSolutions.Data.Migrations/202301101342_AddSelfAssessmentResultDelegateUserID.cs:35]: Down migration is updated as per review comments.

**Review by kevwhitt-hee — APPROVED** [review]

**Pull request merged**